src/Testing/Console/TestConsoleHelper.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Testing\Console;
+
+use PHPUnit\Framework\Assert;
+
+final readonly class TestConsoleHelper
+{
+    public function __construct(
+        private TestConsoleOutput $output,
+    ) {
+    }
+
+    public function assertContains(string $text): self
+    {
+        Assert::assertStringContainsString(
+            $text,
+            $this->output->getTextWithoutFormatting(),
+            sprintf('Failed to assert that console output included text: %s.', $text)
+        );
+
+        return $this;
+    }
+
+    public function assertContainsFormattedText(string $text): self
+    {
+        Assert::assertStringContainsString(
+            $text,
+            $this->output->getTextWithFormatting(),
+            sprintf('Failed to assert that console output included formatted text: %s.', $text)
+        );
+
+        return $this;
+    }
+
+    public function assertContainsError(string $line): self
```

Comment:
I like these assertions, but what about formatting? Right now it are the formatted error lines that are checked, which doesn't make much sense. Alternatively, we can store the unformatted lines, but what if we want to test on formatted error messages then. Adding `assertContainsFormattedError` and `assertContainsUnformattedError` also seems a bit overkill…

I think we need a way to get whatever output, and then optionally call `unformat` or something on it. IDK what exactly that should look like though…

Reply:
It's a good point. What I had been playing around with _prior_ to the console changes was passing two parameters (string and variadic styles) like you had for the formatter. If no styles were passed, it would do an unformatted test.

I think my inclination now is that _all_ methods expect the developer to pass the string as they would have passed it to output (so they are dictating the styling) and we have a single `assertContainsText` method that does an unformatted comparison if we are just looking for a single string.

Thoughts?

Reply:
> I think my inclination now is that all methods expect the developer to pass the string as they would have passed it to output (so they are dictating the styling) and we have a single assertContainsText method that does an unformatted comparison if we are just looking for a single string.

I don't really understand, could you give an example?